Marlin/src/feature/backlash.cpp

```diff
@@ -87,22 +85,20 @@ void Backlash::add_correction_steps(const int32_t &da, const int32_t &db, const
   #endif
   last_direction_bits ^= changed_dir;
 
-  if (correction == 0) return;
+  bool nothing_to_do = !correction;
+  for (uint8_t axis = 0; nothing_to_do && axis < LINEAR_AXES; axis++)
```

Comment:
Shouldn't this rather be… `for (uint8_t axis = 0; !nothing_to_do && axis < LINEAR_AXES; axis++)` ?

Reply:
I am pretty sure I had it right (head is hurting though!).

`nothing_to_do` means that all the values are zero. i.e. `correction == 0 && residual_error[X_AXIS] == 0 && ...`

[36dc96e](https://github.com/MarlinFirmware/Marlin/pull/23826/commits/36dc96e15921355e96bd9a0b3936a9d3cbcaccb2) is equivalent to `correction == 0 || residual_error[X_AXIS] == 0 || ...`

Reply:
As it is, the code definitely does the wrong thing because I found in testing that `if (correction == 0) return;` prevents the previously applied steps from being unapplied after `M425 F0`.

Reply:
The code I'm commenting on here would never change the state of `nothing_to_do` since it is required to be set for the loop to continue at all. I got the sense that what you want is to not only return when `correction == 0` but then also return if all of the `residual_error` values are equal to zero. That is what the code now expresses, so please make a correction if I have something backward.

Reply:
Based on your description (and with help from `types.h`) I have simplified it to…
```cpp
if (!correction && !residual_error) return;
```